commons-db/src/main/java/com/palantir/nexus/db/sql/SQLString.java

```diff
@@ -165,13 +165,15 @@ public static boolean isQueryRegistered(String key) {
      * @param dbType Look for queries registered with this override first
      * @return a SQLString object representing the stored query
      */
-    @SuppressWarnings("GuardedByChecker")
     static FinalSQLString getByKey(final String key, DBType dbType) {
         assert isValidKey(key) : "Keys only consist of word characters"; // $NON-NLS-1$
         assert registeredValues.containsKey(key) || registeredValuesOverride.containsKey(key)
                 : "Couldn't find SQLString key: " + key + ", dbtype " + dbType; // $NON-NLS-1$ //$NON-NLS-2$
 
-        FinalSQLString cached = cachedKeyed.get(key);
+        FinalSQLString cached;
+        synchronized (cacheLock) {
+            cached = cachedKeyed.get(key);
```

Comment:
It's not clear that this `get` invocation must occur holding `cacheLock`, but it's also not clear that the lock buys us anything on `volatile` reads/writes either, since only access and modify operations are locked without any state checks (the `runWithCacheLock` method appears unused internally).

Perhaps we can remove `runWithCacheLock` and drop the lock, relying on the volatile reference for ordering?

Reply:
The whole `cachedKeyed`/`cacheLock` here is a bit of a mess. 
In a follow on PR after we unblock gradle-baseline, I do think we can probably remove `runWithCacheLock` (and possibly the whole `cachedKeyed`, `cacheLock`, `setCachedKeyed`, `getCachedKeyed`) as I don't see any consumers of these APIs.


Note that just removing the `synchronized (cacheLock)` from the volatile read still triggers the `GuardedBy` error (as expected), so I refactored into a separate method to explicitly do volatile read & suppress (also handle possible NPE).

```
error: [GuardedBy] This access should be guarded by 'SQLString.cacheLock', which is not currently held
        FinalSQLString cached = cachedKeyed.get(key);
```